include/nbl/video/IGPUDescriptorSet.h

```diff
@@ -30,65 +29,72 @@ class NBL_API IGPUDescriptorSet : public asset::IDescriptorSet<const IGPUDescrip
 		using base_t = asset::IDescriptorSet<const IGPUDescriptorSetLayout>;
 
 	public:
-		IGPUDescriptorSet(core::smart_refctd_ptr<const ILogicalDevice>&& dev, core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& _layout, core::smart_refctd_ptr<IDescriptorPool>&& pool)
-			: base_t(std::move(_layout)), IBackendObject(std::move(dev)), m_version(0ull), m_pool(std::move(pool))
-		{
-            allocateDescriptors();
-
-            for (auto i = 0u; i < asset::EDT_COUNT; ++i)
-            {
-                // There is no descriptor of such type in the set.
-                if (m_descriptorStorageOffsets.data[i] == ~0u)
-                    continue;
+        IGPUDescriptorSet(core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& _layout, core::smart_refctd_ptr<IDescriptorPool>&& pool, IDescriptorPool::SDescriptorOffsets&& offsets);
 
-                const auto type = static_cast<asset::E_DESCRIPTOR_TYPE>(i);
+        inline uint64_t getVersion() const { return m_version.load(); }
```

Comment:
If we implement https://github.com/Devsh-Graphics-Programming/Nabla/pull/345/files/8fbc87079edc5109a9c916eeb7ea7c0eff9ce128..b3719ddb1377d8af8f8d5f4dd112d3c844394553#r1116359755 then instead of `m_version` we could keep an `m_isZombie` and let `IDescriptorSet` set that via friendship.

And this public method wouldn't be needed.